<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
  <link rel="icon" href="../favicon.ico" />
  <link rel="stylesheet" href="../style.css">
  <title>警告: URI.escape は廃止されました | Techioz</title>
  <style type="text/css">code{white-space: pre;}</style>
  <script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-1637770322917058"
     crossorigin="anonymous"></script>
  <!-- Google tag (gtag.js) -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-GV53GJEY2E"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'G-GV53GJEY2E');
  </script>
<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "Article",
  "headline": "警告: URI.escape は廃止されました",
  "author": {
    "@type": "Person",
    "address": "",
    "email": null,
    "identifier": "masyumaroking",
    "name": "masyumaroking"
  },
  "publisher": {
    "@type": "Person",
    "name": "masyumaroking"
  }
}
</script>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
  <header>
    <div class="container">
        <h1>Techioz Blog</h1>
        <nav>
            <ul>
                <li><a href="#">Home</a></li>
                <li><a href="#">Articles</a></li>
                <li><a href="#">About</a></li>
                <li><a href="#">Contact</a></li>
            </ul>
        </nav>
    </div>
  </header>
  <main>
    <div class="container">
        <article>
                      <h2>警告: URI.escape は廃止されました</h2>
                    <h2 id="概要">概要</h2>
                    <p>こんにちは、クイックブックから Oauth2
                    アクセスを取得しようとしていますが、レールサーバーからのコマンドプロンプトでこの警告が表示されますが、何も起こりません。サーバーは同じ警告を読み続けるだけです。</p>
                    <p>これはエラーを引き起こす行です (gem
                    内にあります)。</p>
                    <pre><code>    def escape(value)
      URI::escape(value.to_s, OAuth::RESERVED_CHARACTERS)
    rescue ArgumentError
      URI::escape(value.to_s.force_encoding(Encoding::UTF_8), OAuth::RESERVED_CHARACTERS)
    end
</code></pre>
                    <p>このエラーに関する他の苦情を見て、URI::escape()
                    コマンドを CGI::escape()
                    に置き換えようとしましたが、このエラーが発生しました。</p>
                    <p>誰かがこのエラーを解決し、機能する代替手段を見つける方法を教えてもらえますか?
                    私の Ruby バージョン: Ruby 2.7.2p137
                    私のRailsバージョン: Rails 6.1.4 ありがとう。</p>
                    <h2 id="解決策">解決策</h2>
                    <p>URI.escape
                    の動作する実装だけが必要な場合は、それを自分で再実装できます。</p>
                    <pre><code># config/initializers/uri.rb

module URI
  def self.escape(*args)
    DEFAULT_PARSER.escape(*args)
  end
end
</code></pre>
                    <p>これにより、警告が効果的に削除され、エスケープ呼び出しがデフォルトのパーサーに委任されます。これは
                    Ruby 3 でも機能します。</p>
                    <p>これは警告を消すための単なる簡単なパッチであることに注意してください。長期的には、古くなった
                    gem の代替品を見つける必要があります。</p>
        </article>
    </div>
  </main>
  <footer>
    <div class="container">
        <p>&copy; 2024 Techioz. All rights reserved.</p>
    </div>
  </footer>
</body>
</html>